src/demo/state.js

```diff
@@ -28,14 +28,22 @@ export const getState = function() {
 };
 
 export const setState = function(newState) {
-  state = {...state, ...newState};
+  return setStateInternal({...state, ...newState});
+};
+
+export const setInitialState = function(newState) {
+  return setStateInternal({...initialState, ...newState});
+};
```

Comment:
@madelynkasula - @cpirich explained to me that this allows us to reinitialise state multiple times in a single app instance, which allows the locally-served version of ml-activities to switch between app modes without doing a page reload.